static/pushNotification.js

```diff
@@ -0,0 +1,35 @@
+
+function pushNotification(task, path){
+  let userName = /[^@]*/.exec(task.user.name);
+  let body     = `${userName}'s deploy ${task.status}`;
+  let {hostname, protocol,port} = location;
+
+  if(hostname == "localhost"){
+    var url  = `${protocol}//${hostname}:${port}${path}`;
+  }else{
+    var url  = `${protocol}//${hostname}${path}`;
+  }
```

Comment:
Note that `location.origin` exists, so you can replace these 5 lines of code with:

```javascript
let {origin} = location;
var url = `${origin}${path}`;
```